File: Code/14.DataBinding 2/app/src/main/java/com/github/aprofromindia/databinding/MainActivity.java
package com.github.aprofromindia.databinding;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.github.aprofromindia.databinding.databinding.ActivityMainBinding;
import com.github.aprofromindia.databinding.viewModels.MainViewModel;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil
                .setContentView(this, R.layout.activity_main);
        MainViewModel vm = new MainViewModel(this);
        binding.setViewModel(vm);
    }
}

File: Code/15 - Kotlin Demo/app/src/main/java/com/github/aprofromindia/demoapplication/MainActivity.kt
package com.github.aprofromindia.demoapplication

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener{
         Toast.makeText(this@MainActivity, "Hello Aperto Engineering", Toast.LENGTH_LONG).show()
        }

        actBtn.setOnClickListener{
            startActivity(Intent(this@MainActivity, DetailActivity::class.java))
        }

        snkButton.setOnClickListener{
            Snackbar.make(it, "Hello Snackbar", Snackbar.LENGTH_LONG).show()
        }

        textView.text = "Something different"
    }
}
